dhis-2/dhis-services/dhis-service-administration/src/main/java/org/hisp/dhis/merge/category/CategoryOptionMergeService.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2004-2024, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.merge.category;
+
+import com.google.common.collect.ImmutableList;
+import jakarta.persistence.EntityManager;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import javax.annotation.Nonnull;
+import javax.annotation.PostConstruct;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.hisp.dhis.category.CategoryOption;
+import org.hisp.dhis.category.CategoryService;
+import org.hisp.dhis.common.UID;
+import org.hisp.dhis.dataelement.DataElement;
+import org.hisp.dhis.feedback.MergeReport;
+import org.hisp.dhis.merge.MergeParams;
+import org.hisp.dhis.merge.MergeRequest;
+import org.hisp.dhis.merge.MergeService;
+import org.hisp.dhis.merge.MergeValidator;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * Main class for a {@link CategoryOption} merge.
+ *
+ * @author david mackessy
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+public class CategoryOptionMergeService implements MergeService {
+
+  private final CategoryService categoryService;
+  private final CategoryOptionMergeHandler categoryOptionMergeHandler;
+  private final MergeValidator validator;
+  private final EntityManager entityManager;
+  private ImmutableList<MetadataMergeHandler> metadataMergeHandlers;
```

Comment:
I think we want to stop using `ImuutableList`. You can make a de-facto immutable list using `List.copyOf`

Reply:
I will change this 👍. `List.of` will do.